tests/test_algorithm.py

```diff
@@ -1965,3 +1966,41 @@ def analyze(context, perf):
                                 analyze=analyze)
         results = algo.run(self.panel)
         self.assertIs(results, self.perf_ref)
+
+
+class TestRemoveData(TestCase):
+    """
+    tests if futures data is removed after expiry
+    """
+    def setUp(self):
+        dt = pd.Timestamp('2015-01-01', tz='UTC')
+        metadata = {0: {'asset_type': 'future', 'symbol': 'X',
+                        'expiration_date': dt + timedelta(days=4)},
+                    1: {'asset_type': 'future', 'symbol': 'Y',
```

Comment:
I don't think 'asset_type' is necessary as the type is now implied from the argument name to write_data